### ISSUE TITLE
How to get actual probabilties for actions given a state for policy gradient based methods. [question]

### ISSUE THREAD

**Issue by mynkpl1998** [ray-project/ray#4572]
Hi, I want to see what probabilities my agent predicting in the case of discrete action space given a state input to the network. Is there any way to get access to those values. I looked at the compute_action function but it doesn't return probability distribution.

**Comment by ericl** [comment]
Most policies emit "action_prob" as an extra info fetch of computing actions. However you'll need to call the low-level batch compute_actions() method directly. For example:

`agent.get_policy().compute_actions()` where the args to compute_actions is defined here: https://github.com/ray-project/ray/blob/master/python/ray/rllib/evaluation/policy_graph.py#L48

Note that DQN

**Comment by ericl** [comment]
After https://github.com/ray-project/ray/pull/4556 is merged, you can also call `compute_action(..., full_fetch=True)` to retrieve these values, which will be a bit more convenient.

**Comment by mynkpl1998** [comment]
The following code worked for me. Thanks for the suggestion @ericl 
```
self.agent = PPOAgent(env="tsim-v0", config=exp_config[exp_name]["config"])
self.agent.restore(checkpoint_path)

def getAction(self, observation):
  state = []
  preprocessed = self.agent.local_evaluator.preprocessors["default"].transform(observation)
  filtered_obs = self.agent.local_evaluator.filters['default'](preprocessed, update=False)
  result = self.agent.get_policy('default').compute_single_action(filtered_obs, state, None, None, None, clip_actions=self.agent.config["clip_actions"])
  action = result[0]
  probs = softmax(result[2]["logits"]) # imported from scipy.special
  return action, probs
```
